src/main/java/net/dv8tion/jda/core/requests/RestAction.java

```diff
@@ -615,18 +655,27 @@ public T completeAfter(long delay, TimeUnit unit)
      */
     public ScheduledFuture<?> queueAfter(long delay, TimeUnit unit, Consumer<T> success, Consumer<Throwable> failure, ScheduledExecutorService executor)
     {
-        Args.notNull(executor, "Scheduler");
-        Args.notNull(unit, "TimeUnit");
+        Checks.notNull(executor, "Scheduler");
+        Checks.notNull(unit, "TimeUnit");
         return executor.schedule(() -> queue(success, failure), delay, unit);
     }
 
-    protected void finalizeData() { }
+    protected RequestBody finalizeData() { return data; }
+    protected Route.CompiledRoute finalizeRoute() { return route; }
+    protected CaseInsensitiveMap<String, String> finalizeHeaders() { return headers == null ? null : new CaseInsensitiveMap<>(headers); } // maps are mutable 
```

Comment:
I reverted this to only have the method which returns a map of headers, the field addition as of now is not needed and a pointless change. In the case of this being needed we can adjust this as fits.